src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeArgumentInference.vb

```diff
@@ -2094,6 +2094,10 @@ HandleAsAGeneralExpression:
                     Dim delegateParams As ImmutableArray(Of ParameterSymbol) = invokeMethod.Parameters
                     Dim minCount As Integer = Math.Min(lambdaParams.Length, delegateParams.Length)
 
+                    If lambdaParams.Length > minCount Then
+                        Return True
+                    End If
```

Comment:
If this fix will work, we will rename some of variables here and make it more elegant.